docs/semgrep-ci/manual-ci-setup.md

```diff
@@ -0,0 +1,414 @@
+
+<!-- ![Steps to integrate Semgrep in CI manually](images/image1.png "image_tooltip") -->
+*Figure 1. Steps to integrate Semgrep in CI manually.*
+
+This guide defines a job or CI job as a script executed within a certain environment and managed by a CI provider.
+
+By refining a job's parameters, you are able to achieve the following goals:
+
+* **Run Semgrep on a schedule.** Run full scans on mainline branches at a least intrusive time on developer teams.
+* **Run Semgrep with custom rules.** Apply rules specific to your organization's business goals and coding conventions.
+* **Run Semgrep when an event triggers.** Run Semgrep when a pull or merge request (PR or MR) is created. These event triggers or event hooks are dependent on your CI provider. 
+* **Run Semgrep on relevant files and blocks of code.** Configure Semgrep to ignore files and folders such as test files, configuration files, and files from other vendors.
+* **Pass a Semgrep CI job even when scans report findings.** By default, manual configurations fail when any finding is detected. Configure Semgrep to pass CI jobs even when findings are reported (a **fail open** state).
+
+## Limitations of manually configured Semgrep CI scans 
+
+* Findings are not tiered based on severity and actions cannot be undertaken based on severity. This means that either any finding will cause the job to fail, or any finding still allows the job to pass.
+* Findings are dumped to a log and are not tracked over time, so there is no record of a finding's state, such as opened, closed, or ignored.
+
+## Setting up the CI job
+
+### Introduction
+
+ `semgrep ci` is the command used to run Semgrep in a CI environment. In most cases, this is the recommended command to run in the CI job. It is a subset of the `semgrep scan` command. Features of this command include:
+ 
+ * `semgrep ci` is git-aware. This means it is able to detect branches and git states. 
+ * `semgrep ci` makes use of environment variables to configure its behavior. 
+ * `semgrep ci` performs a diff-aware scan by default. This means it only scans changes in files when run on a pull or merge request.
+ * `semgrep ci` can be run on a local `.git` repository at any time to test its behavior before running it within a CI environment.
+
+:::note
+An alternative method of running Semgrep in your CI is to run the `semgrep scan` command, passing rules to scan, for example `semgrep scan --config auto` which performs a full scan of the repository every time the job is run.
+:::
+
+### GitHub Actions
+
+To add Semgrep into your GitHub Actions pipeline, do the following steps.
+
+1. Create a `semgrep.yaml` configuration file to add a Semgrep command as part of your pipeline. Refer to the [GitHub Actions code snippet](#github-actions-code-snippet).
+2. Optional: You can also review the [GitHub Actions workflow syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) to help with editing the configuration file.
```

Comment:
This is for an entirely manual set up (not sure if that was clear? Let me know.) I am trying to only include environment variables relevant to manual non-App setups.

There's this part in our giant sample config where "If you are not using Semgrep App, uncomment <ThisEnvVar>, but if you are using App, uncomment this <TokenVar>and comment out <this bunch of vars>. It's easy to comment out the wrong thing inadvertently, so instead of making the user puzzle it out, I separate out manual and App-connected CI jobs, including the config file. I hope this makes sense?

Reply:
I have kept the informational comments. Resolving this.